crates/typst/src/geom/color.rs

```diff
@@ -740,3 +2249,69 @@ mod tests {
         test("14B2AH", "color string contains non-hexadecimal letters");
     }
 }
+
+/// A 32-bit float that implements `Eq`, `Ord` and `Hash`.
```

Comment:
This could be moved to the `scalar.rs` file

Reply:
Mind you, I made it purposefully simpler and less complete since it is only used for colors.